sphinx/builders/htmlhelp.py

```diff
@@ -207,6 +209,9 @@ def init(self):
 
     def open_file(self, outdir, basename, mode='w'):
         # type: (unicode, unicode, unicode) -> IO
+        if mode != 'w':
+            warnings.warn('The mode argument of HTMLHelpBuilder.open_file() is deprecated.',
+                          RemovedInSphinx40Warning)
```

Comment:
Instead of this, it would be better to give `mode` option explicitly to me. I think "explicit" is important concept for Python programs.
So -1 for this.